rockcraft/extensions/gunicorn.py

```diff
@@ -0,0 +1,340 @@
+                "flask-framework/install-app part to start with flask/app"
+            )
+        if not user_prime:
+            user_prime = [
+                f"flask/app/{f}"
+                for f in (
+                    "app",
+                    "app.py",
+                    "migrate",
+                    "migrate.sh",
+                    "migrate.py",
+                    "static",
+                    "templates",
+                )
+                if (self.project_root / f).exists()
+            ]
+        return user_prime
+
+    def _check_wsgi_path(self):
+        """Ensure the extension can infer the WSGI path of the Flask application."""
+        app_file = self.project_root / "app.py"
+        if not app_file.exists():
+            raise ExtensionError(
+                "flask application can not be imported from app:app, "
+                "no app.py file found in the project root"
+            )
+        if not self.has_global_variable(app_file, "app"):
+            raise ExtensionError(
+                "flask application can not be imported from app:app, "
+                "no variable named app in app.py"
+            )
+
+    @override
+    def check_project(self):
+        """Ensure this extension can apply to the current rockcraft project."""
+        if not (self.project_root / "requirements.txt").exists():
+            raise ExtensionError(
+                "missing requirements.txt file, "
+                "flask-framework extension requires this file with flask specified as a dependency"
+            )
+        if not self.yaml_data.get("services", {}).get("flask", {}).get("command"):
+            self._check_wsgi_path()
+
+
+class DjangoFramework(_GunicornBase):
+    """An extension for constructing Python applications based on the Django framework."""
+
+    @property
+    def name(self):
+        """Return the normalized name of the rockcraft project."""
+        return self.yaml_data["name"].replace("-", "_").lower()
+
+    @property
+    def default_wsgi_path(self):
+        """Return the default wsgi path for the Django project."""
+        return f"{self.name}.wsgi:application"
+
+    @property
+    @override
+    def wsgi_path(self) -> str:
+        """Return the wsgi path of the wsgi application."""
+        return self.default_wsgi_path
+
+    @property
+    @override
+    def framework(self) -> str:
+        """Return the wsgi framework name, e.g. flask, django."""
+        return "django"
+
+    @override
+    def gen_install_app_part(self) -> Dict[str, Any]:
+        """Return the prime list for the Flask project."""
+        if "django-framework/install-app" not in self.yaml_data.get("parts", {}):
+            return {
+                "plugin": "dump",
+                "source": ".",
+                "organize": {self.name: "django/app"},
+                "stage": ["django/app"],
+                "prime": ["django/app"],
+            }
```

Comment:
I am not very confident about this. Is there a better way to achieve the same effect? I tried this, but it doesn't work.

```yaml
django-framework/install-app:
  plugin: dump
  source: "{self.name}"
  organize: 
    ".": django/app
```

Reply:
Does the wildcard patten here matches files start with dot `.`?

Reply:
hmm I'm not sure. If it doesn't, then maybe this will:

```
  organize:
    "*": django/app
    ".*": django/app

Reply:
I did some experiment and found that `organize` indeed uses `glob` rules (which don't match files starting with `.`) instead of `fnmatch` rules. Here is the final working version (the target directory needs to end with `/`):

```yaml
  organize:
    "*": django/app/
    ".*": django/app/
```

Thank you for your suggestion! updated.